packages/shared/useArraySome/index.ts

```diff
@@ -0,0 +1,11 @@
+import type { MaybeComputedRef } from '@vueuse/shared'
+import { resolveUnref } from '@vueuse/shared'
+import type { ComputedRef } from 'vue-demi'
+import { computed } from 'vue-demi'
+
+export function useArraySome<T>(
+  list: MaybeComputedRef<MaybeComputedRef<T>[]>,
+  fn: (element: T, index: number, array: T[]) => boolean,
+): ComputedRef<boolean> {
+  return computed(() => resolveUnref(list).map(i => resolveUnref(i)).some(fn))
```

Comment:
Maybe we could unref the item when calling the function to avoid creating a new array

Reply:
Agree. Thanks for your advice